backend/services/notification_service.py

```
@@ -0,0 +1,354 @@
+            created_count = 0
+            for student in students:
+                notification = self.create_assignment_due_notification(
+                    db, student.id, assignment_title, due_date
+                )
+                if notification:
+                    created_count += 1
+            
+            logger.info(f"Created {created_count} assignment due notifications for cohort {cohort_id}")
+            return created_count
+            
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)